pkg/validators/validator_001_default_channel.go

```diff
@@ -0,0 +1,91 @@
+package validators
+
+import (
+	"fmt"
+
+	"github.com/mt-sre/addon-metadata-operator/api/v1alpha1"
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+)
+
+// ValidateDefaultChannel validates whether the 'defaultChannel' provided under an addon.yaml also exists under 'channels' field
+func ValidateDefaultChannel(metabundle utils.MetaBundle) (bool, string, error) {
+	defaultChannel := metabundle.AddonMeta.DefaultChannel
+	channels := metabundle.AddonMeta.Channels
+
+	if success, failureMsg := isPartOfEnum(defaultChannel); !success {
+		return false, failureMsg, nil
+	}
+	// to be deprecated - only used for legacy builds
+	if success, failureMsg := isListedInChannels(channels, defaultChannel); !success {
+		return false, failureMsg, nil
+	}
+
+	// if success, failureMsg, err := matchesDefaultChannelAnnotations(metabundle.Bundles, defaultChannel); !success {
+	// 	return false, failureMsg, err
+	// }
+	return true, "", nil
+}
+
+func isPartOfEnum(defaultChannel string) (bool, string) {
+	enum := map[string]struct{}{
+		"alpha":  {},
+		"beta":   {},
+		"stable": {},
+		"edge":   {},
+		"rc":     {},
+	}
+	if _, ok := enum[defaultChannel]; !ok {
+		return false, fmt.Sprintf("The defaultChannel '%v' is not part of the accepted values: alpha, beta, stable, edge or rc.", defaultChannel)
+	}
+	return true, ""
+}
+
+// TODO - deprecate this when we remove legacy builds
+func isListedInChannels(channels *[]v1alpha1.Channel, defaultChannel string) (bool, string) {
+	// as the Channels field is deprecated, it can be omitted
+	if channels == nil {
+		return true, ""
+	}
+	var channelNames []string
+	for _, channel := range *channels {
+		if channel.Name == defaultChannel {
+			return true, ""
+		}
+		channelNames = append(channelNames, channel.Name)
+	}
+	return false, fmt.Sprintf("The defaultChannel '%v' is not part of the listed channelNames: %v.", defaultChannel, channelNames)
+}
+
+// TODO - (sblaisdo) enable when extraction format is bundles instead of packageManifest
+// if the annotation is not present, make sure the defaultChannel is alpha
+// func matchesDefaultChannelAnnotations(bundles []registry.Bundle, defaultChannel string) (bool, string, error) {
+// 	for _, bundle := range bundles {
+// 		allChannels := []string{bundle.Annotations.Channels, "dontexist"}
+// 		var defaultBundleChannel string
+
+// 		if len(allChannels) == 1 {
+// 			// if a single channel is listed, should match defaultChannel
+// 			defaultBundleChannel = allChannels[0]
```

Comment:
Shouldn't here be a sanity check to confirm that `allChannels[0]` / `defaultBundleChannel` matches `bundle.Annotations.DefaultChannelName` ? 

Or should there be a dedicated check for that?

Reply:
So it's two cases:
1. if you have a single channel listed in `operators.operatorframework.io.bundle.channels.v1`, then you want the `defaultChannel` to match this
2. If you have multiple channels listed in `operators.operatorframework.io.bundle.channels.v1`, then you need the extra annotation `operators.operatorframework.io.bundle.channel.default.v1` matching the `defaultChannel`

Reply:
No I meant that say, there's only single channel under `allChannels`, say, it is "alpha". In that case, say, even the `bundle.Annotations.defaultChannel` is specified (although it's optional in this case), and say, it is "beta".

Then that's an incoherence right? And a validation error should be reported to either suggest to remove the `bundle.Annotations.defaultChannel` field because it is not needed as there's anyway only one channel under `bundle.Annotations.Channels` or to ensure that the `bundle.Annotations.defaultChannel` matches the single only channel under `bundle.Annotations.Channels`.

But I believe a separate validator for this check would be more suitable. Let me create an item for that in google sheet, create a JIRA ticket and wrap it up :)